client/src/components/Multiselect.tsx

```diff
@@ -1,712 +1,348 @@
+          IndicatorSeparator: () => null,
+          IndicatorsContainer: () => null,
+          ClearIndicator: () => null,
+          SelectContainer: SelectContainer,
+          Option: CustomOption,
+          Input: CustomInput,
+        }}
+        // Custom props passed through SelectProps to composable components
+        removeValue={removeValue}
+        onApply={onApply}
+        setSelections={setSelections}
+        showAllSelections={showAllSelections}
+        setShowAllSelections={setShowAllSelections}
+        previewSelectedNum={previewSelectedNum || 5}
+        {...props}
+      />
+
+      {infoAlert}
+
+      <button
+        className="button is-primary"
+        aria-label={i18n._(t`Apply selections and get results`)}
+        onClick={handleApply}
+        // By default, when multiselect option list is open mouseDown outside
+        // closes the list, so because the Apply button is below, if you try to
+        // click apply while it's open it moves on you before you can mouseUp.
+        onMouseDown={handleApply}
+        onTouchStart={handleApply}
+      >
+        <Trans>Apply</Trans>
+      </button>
+    </div>
+  );
+}
 
-  renderOptionList() {
-    const { id, style, emptyRecordMsg, loading, loadingMessage = "loading..." } = this.props;
-    const { options, highlightOption } = this.state;
-    if (loading) {
-      return (
-        <ul className={`optionContainer`} style={style["optionContainer"]}>
-          {typeof loadingMessage === "string" && (
-            <span style={style["loadingMessage"]} className={`notFound`}>
-              {loadingMessage}
-            </span>
-          )}
-          {typeof loadingMessage !== "string" && loadingMessage}
-        </ul>
-      );
-    }
+function SelectedValuesContainer<
+  Option,
+  IsMulti extends boolean = true,
+  GroupType extends GroupBase<Option> = GroupBase<Option>
+>({ isDisabled, getValue, ...props }: ContainerProps<Option, IsMulti, GroupType>) {
+  const {
+    getOptionValue,
+    removeValue,
+    classNamePrefix,
+    showAllSelections,
+    setShowAllSelections,
+    previewSelectedNum,
+  } = props.selectProps as Props<Option, IsMulti, GroupType> & CustomSelectProps;
+
+  const getKey = (opt: Option, index: number) =>
+    `${getOptionValue ? getOptionValue(opt) : "option"}-${index}`;
+
+  const toMultiValue = (opt: Option, index: number) => {
     return (
-      <ul
-        tabIndex="-1"
-        role="listbox"
-        className={`optionContainer`}
-        style={style["optionContainer"]}
-        aria-activedescendant={`${id || "option"}-item-${highlightOption}`}
+      <components.MultiValue
+        getValue={getValue}
+        {...props}
+        components={{
+          Container: components.MultiValueContainer,
+          Label: components.MultiValueLabel,
+          Remove: CustomMultiValueRemove,
+        }}
+        // TODO: Can't know when it's "focused" via arrow key navigation, it's not using ::focus,
```

Comment:
thoughts on making github issues for the things that aren't breaking errors but would be nice to fix? unsure if that's a good use of our time but i dont know if these non-breaking errors are being kept track of anywhere. they don't feel urgent enough to be addressed right now in VQA, but I think we do need a visible log of them